src/java/test/org/apache/zookeeper/client/StaticHostProviderTest.java

```diff
@@ -110,6 +110,71 @@ public void testLiteralIPNoReverseNS() throws Exception {
         }
     }
 
+    @Test
+    public void testReresolutionAfterNoConnection() throws Exception {
+        byte size = 1;
+        ArrayList<InetSocketAddress> addresses = new ArrayList<InetSocketAddress>(size);
+
+        addresses.add(InetSocketAddress.createUnresolved("issues.apache.org", 1234));
+        addresses.add(InetSocketAddress.createUnresolved("wiki.apache.org", 1234));
+
+        StaticHostProvider hostProvider = new StaticHostProvider(addresses);
+        assertEquals(2, hostProvider.size());
+
+        // check that first next results in no re-resolution
+        hostProvider.next(0);
+        assertEquals(0, hostProvider.getAddedAddresses());
+        assertEquals(0, hostProvider.getRemovedAddresses());
+        // check that second next _does_ result in re-resolution
+        hostProvider.next(0);
+        assertEquals(1, hostProvider.getAddedAddresses());
+        assertEquals(1, hostProvider.getRemovedAddresses());
+    }
+
+    @Test
+    public void testNoReresolutionAfterConnection() throws Exception {
+        byte size = 1;
+        ArrayList<InetSocketAddress> addresses = new ArrayList<InetSocketAddress>(size);
+
+        addresses.add(InetSocketAddress.createUnresolved("issues.apache.org", 1234));
+        addresses.add(InetSocketAddress.createUnresolved("wiki.apache.org", 1234));
+
+        StaticHostProvider hostProvider = new StaticHostProvider(addresses);
+        assertEquals(2, hostProvider.size());
+
+        // check that first next results in no re-resolution
+        hostProvider.next(0);
+        assertEquals(0, hostProvider.getAddedAddresses());
+        assertEquals(0, hostProvider.getRemovedAddresses());
+        // Call onConnected
+        hostProvider.onConnected();
+        // check that second next results in no re-resolution
+        hostProvider.next(0);
+        assertEquals(0, hostProvider.getAddedAddresses());
+        assertEquals(0, hostProvider.getRemovedAddresses());
+    }
+
+    @Test
+    public void testReresolutionMultiAddress() throws Exception {
+        byte size = 1;
+        ArrayList<InetSocketAddress> addresses = new ArrayList<InetSocketAddress>(size);
+
+        addresses.add(InetSocketAddress.createUnresolved("issues.apache.org", 1234));
+        addresses.add(InetSocketAddress.createUnresolved("apache.org", 1234));
+
+        StaticHostProvider hostProvider = new StaticHostProvider(addresses);
+        assertEquals(3, hostProvider.size());
+
+        // check that first next results in no re-resolution
+        hostProvider.next(0);
+        assertEquals(0, hostProvider.getAddedAddresses());
+        assertEquals(0, hostProvider.getRemovedAddresses());
+        // check that second next _does_ result in re-resolution
+        hostProvider.next(0);
+        assertTrue(hostProvider.getAddedAddresses() > 0);
```

Comment:
So, `addedAddresses > 0` indicates reresolution? Should we not update the variable name to indicate that?

Reply:
Lifted var names straight from the upstream.